runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1982,20 +1981,20 @@ MM_CopyForwardScheme::copy(MM_EnvironmentVLHGC *env, MM_AllocationContextTarok *
 
 				memcpy((void *)destinationObjectPtr, forwardedHeader->getObject(), objectCopySizeInBytes);
 
-				forwardedHeader->fixupCopiedObject(destinationObjectPtr);
+				forwardedHeader->fixupForwardedObject(destinationObjectPtr);
 
-				if(_extensions->objectModel.isIndexable(destinationObjectPtr)) {
+				if(objectModel->isIndexable(destinationObjectPtr)) {
 					updateInternalLeafPointersAfterCopy((J9IndexableObject *)destinationObjectPtr, (J9IndexableObject *)forwardedHeader->getObject());
 				}
 
 				/* IF the object has been hashed and has not been moved then we must store the previous
 				 * address into the hashcode slot at hashcode offset.
 				 */
 				if (doesObjectNeedHash) {
-					UDATA hashOffset = forwardedHeader->getHashcodeOffset(_javaVM, destinationObjectPtr);
+					UDATA hashOffset = objectModel->getHashcodeOffset(destinationObjectPtr);
```

Comment:
Also logic that eventually should be reused from ObjectModel though fixupForwardedObject(). Although we probably slightly differ how we set moved bit in Standard GCs vs Balanced GC, so we would possibly need another variant.